internal/mapping.go

```diff
@@ -145,22 +166,20 @@ func GetSpannerKeyName(srcKeyName string, schemaForeignKeys map[string]bool) str
 	if srcKeyName == "" {
 		return ""
 	}
-	spKeyName, _ := FixName(srcKeyName)
-	if _, found := schemaForeignKeys[spKeyName]; found {
-		// spKeyName has been used before.
-		// Add unique postfix: use number of foreign keys so far.
-		// However, there is a chance this has already been used,
-		// so need to iterate.
-		id := len(schemaForeignKeys)
-		for {
-			c := spKeyName + "_" + strconv.Itoa(id)
-			if _, found := schemaForeignKeys[c]; !found {
-				spKeyName = c
-				break
-			}
-			id++
-		}
-	}
-	schemaForeignKeys[spKeyName] = true
-	return spKeyName
+	return getUniqueKeyName(srcKeyName, schemaForeignKeys)
+}
+
+// GetSpannerKeyName maps source index key name to
+// legal Spanner index key name
+// We need to make sure
+// of the following things:
+// a) the new index key name is legal
+// b) the new index key name doesn't clash with other Spanner
+//    index key names
+// Note that index key constraint names in Spanner have to be globally unique
+// (across the database). But in some source databases, such as MySQL,
+// they only have to be unique for a table. Hence we must map each source
+// constraint name to a unique spanner constraint name.
+func GetSpannerIndexKeyName(srcKeyName string, schemaIndexKeys map[string]bool) string {
```

Comment:
Consider renaming these functions as following:

func GetSpannerKeyName(srcKeyName string, schemaForeignKeys map[string]bool) string
->
func ToSpannerForeignKey(srcForeignKey string, spannerForeignKeys map[string]bool) string

func GetSpannerIndexKeyName(srcKeyName string, schemaIndexKeys map[string]bool) string
->
func ToSpannerIndexKey(srcIndexKey string, spannerIndexKeys map[string]bool) string

func GetSpannerTable(conv *Conv, srcTable string) (string, error) 
->
func ToSpannerTable(srcTable string, conv *Conv) (string, error) {

func GetSpannerCol(conv *Conv, srcTable, srcCol string, mustExist bool) (string, error) {
->
func ToSpannerCol(srcTable, srcCol string, mustExist bool, conv *Conv) (string, error)

Motivation:
1. You can drop Name suffix e.g., srcKeyName -> srcIndexKey, since it's clear from the string type that it's the name and is consistent with the other argument i.e., schemaIndexKeys
2. srcIndexKey provides more info about the input argument compared to srcKeyName
3. To make function definitions consistent in this file.
4. Good idea to move conv *Conv to being the last argument to signal that it may be updated within the function.

Reply:
updated names for GetSpannerKeyName and GetSpannerIndexKeyName, but skipping renaming other functions at the moment, as those are used at lot of places. keeping this comment open so that we can revisit this later.